prow/tide/tide.go

```diff
@@ -200,7 +200,7 @@ func byRepoAndNumber(prs []PullRequest) map[string]PullRequest {
 // Note: an empty diff can be returned if the reason that the PR does not match
 // the TideQuery is unknown. This can happen happen if this function's logic
 // does not match GitHub's and does not indicate that the PR matches the query.
-func requirementDiff(pr *PullRequest, q *config.TideQuery) (string, int) {
+func requirementDiff(pr *PullRequest, q *config.TideQuery, bp config.TideBranchProtection) (string, int) {
```

Comment:
I think it would be useful to better separate out the concerns here, so send `optionalContexts sets.String` or else an interface like:
```
type foo interface {
  requiredSubset([]string) []string
}
```

Reply:
I ll wait for you to answer to "don't you think one would still want to prevent merging when one of the non required checks is failing ?If not then, you are right." before fixing this one